pkg/deploy/deploy.go

```diff
@@ -200,35 +200,37 @@ func DeployManifestsFromPath(
 }
 
 func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Unstructured, owner metav1.Object, applicationNamespace, componentName string, enabled bool) error {
-	// Return if resource is of Kind: Namespace and Name: odhApplicationsNamespace
+	// Skip if resource is of Kind: Namespace and Name: applicationNamespace
 	if obj.GetKind() == "Namespace" && obj.GetName() == applicationNamespace {
 		return nil
 	}
 
 	found, err := getResource(ctx, cli, obj)
 
-	// err == nil means found
-	if err == nil {
-		if enabled {
-			// Exception to not update kserve with managed annotation
-			// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
-			if found.GetAnnotations()[annotations.ManagedByODHOperator] == "false" && componentName == "kserve" {
-				return nil
-			}
-			return updateResource(ctx, cli, obj, found, owner, componentName)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			return err
 		}
-		return handleDisabledComponent(ctx, cli, found, componentName)
-	}
-
-	if k8serr.IsNotFound(err) {
-		// Create resource if it doesn't exist and enabled
+		// Create resource if it doesn't exist and component enabled
 		if enabled {
 			return createResource(ctx, cli, obj, owner)
 		}
+		// Skip if resource doesn't exist and component is disabled
 		return nil
 	}
 
-	return err
+	// when resource is found
+	if enabled {
+		// Exception to not update kserve with managed annotation
+		// do not reconcile kserve resource with annotation "opendatahub.io/managed: false"
+		// TODO: remove this exception when we define managed annotation across odh
+		if found.GetAnnotations()[annotations.ManagedByODHOperator] == "false" && componentName == "kserve" {
+			return nil
+		}
+		return updateResource(ctx, cli, obj, found, owner, componentName)
+	}
+	// Delete resource if it exists and component is disabled
+	return handleDisabledComponent(ctx, cli, found, componentName)
```

Comment:
I do not support this change. It had clear logic found/not-found/error (basically, the same switch we discussed some time ago). Now it's again special case.